bundles/org.openhab.core.ui/src/main/java/org/openhab/core/ui/internal/components/UIComponentSitemapProvider.java

```diff
@@ -412,50 +427,78 @@ private void addIconColor(EList<ColorArray> iconColor, UIComponent component) {
-                        condition.setItem(matcher.group("item"));
-                        condition.setCondition(matcher.group("condition"));
-                        condition.setSign(matcher.group("sign"));
-                        condition.setState(matcher.group("state"));
-                        colorArray.eSet(SitemapPackage.COLOR_ARRAY__CONDITIONS, condition);
+                        colorArray.eSet(SitemapPackage.COLOR_ARRAY__ARG, argument);
+                        List<ConditionImpl> conditions = getConditions(conditionsString, component, key);
+                        colorArray.eSet(SitemapPackage.COLOR_ARRAY__CONDITIONS, conditions);
                         color.add(colorArray);
-                    } else {
-                        logger.warn("Syntax error in {} rule '{}' for widget {}", key, sourceColor,
-                                component.getType());
                     }
                 }
             }
         }
     }
 
-    private void addIconRules(EList<IconRule> iconDef, UIComponent component) {
-        if (component.getConfig() != null && component.getConfig().containsKey("icon")) {
-            for (Object sourceIcon : (Collection<?>) component.getConfig().get("icon")) {
-                if (sourceIcon instanceof String) {
-                    Matcher matcher = ICON_PATTERN.matcher(sourceIcon.toString());
-                    if (matcher.matches()) {
+    private void addIconRules(EList<IconRule> icon, UIComponent component) {
+        if (component.getConfig() != null && component.getConfig().containsKey("iconrules")) {
+            Object sourceIcons = component.getConfig().get("iconrules");
+            if (sourceIcons instanceof Collection<?>) {
+                for (Object sourceIcon : (Collection<?>) sourceIcons) {
+                    if (sourceIcon instanceof String) {
+                        String argument = getRuleArgument(sourceIcon.toString());
+                        List<String> conditionsString = getRuleConditions(sourceIcon.toString(), argument);
                         IconRuleImpl iconRule = (IconRuleImpl) SitemapFactory.eINSTANCE.createIconRule();
-                        ConditionImpl condition = (ConditionImpl) SitemapFactory.eINSTANCE.createCondition();
-                        condition.setItem(matcher.group("item"));
-                        condition.setCondition(matcher.group("condition"));
-                        condition.setSign(matcher.group("sign"));
-                        condition.setState(matcher.group("state"));
-                        iconRule.eSet(SitemapPackage.ICON_RULE__CONDITIONS, condition);
-                        iconRule.setArg(matcher.group("arg"));
-                        iconDef.add(iconRule);
-                    } else {
-                        logger.warn("Syntax error in icon rule '{}' for widget {}", sourceIcon, component.getType());
+                        iconRule.eSet(SitemapPackage.ICON_RULE__ARG, argument);
```

Comment:
Using `colorArray.setArg` is probably proper.

Reply:
Done.